### PR TITLE
task(CI): add config scenario classes

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

add config scenario classes

## Testing

Added tests for the following config features that were not big enough to warrent their own feature files 

1: app.isLaunching true after 10 seconds when Launch Duration set to 0

2: Calling Mark Launch Complete when Launch Duration set to 0

3: Setting long launch time

4: Setting short launch time

5: no uncaught errors when AutoNotify = false

6: No Event is sent when the current release stage is not in "notify release stages

7: Forcing uncaught exceptions to be unhandled

8: Discarding An Error Class

9: appType

10: RedactedKeys

11: context

### CODE REVIEW

File: features/csharp/csharp_config.feature [hunk 1]
```diff
@@ -1,165 +1,139 @@
...
-  @skip_webgl
-  Scenario: Report exception from background thread
-    When I run the game in the "BackgroundThreadException" state
+  Scenario: App Type
+    When I run the game in the "AppType" state
     And I wait to receive an error
     Then the error is valid for the error reporting API sent by the Unity notifier
-    And the exception "message" equals "BackgroundThreadException"
+    And the exception "message" equals "AppType"
+    And the event "app.type" equals "custom"
```
Comment (twometresteve) [inline features/csharp/csharp_config.feature:119]: Do AppType and Context really need to be in separate scenarios?

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Config/MarkLaunchComplete.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    public override void Run()
+    {
+        Invoke("DoException", 10);
+    }
+
+    private void DoException()
+    {
+        Bugsnag.MarkLaunchCompleted();
+        Bugsnag.Notify(new System.Exception("MarkLaunchComplete"));
```
Comment (twometresteve) [inline features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Config/MarkLaunchComplete.cs:19]: Even better if you sent one immediately before the `MarkAsCompleted` as well, checking it says isLaunching = true (or do I mean "true"? 😁 ).  Tests are always better if you can demonstrate that taking an action causes something to change.

**Review by twometresteve — APPROVED** [review]
Looks good, just a few suggestions.

File: features/csharp/csharp_config.feature [hunk 3]
```diff
@@ -1,165 +1,139 @@
...
-    And I wait to receive an error
+    And the exception "message" equals "Error 1"
+    And the event "app.isLaunching" equals "true"
+    And I discard the oldest error
+    And the exception "message" equals "Error 2"
+    And the event "app.isLaunching" equals "false"
+  
+  @macos_only
+  Scenario: Set short launch time
+    When I run the game in the "ShortLaunchTime" state
```
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply features/csharp/csharp_config.feature:37]: I'm sure everything will be fine forever and never break 👍

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Config/MarkLaunchComplete.cs (hunk 2 above)
Comment (rich-bugsnag) [reply features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Config/MarkLaunchComplete.cs:19]: done 👍

File: features/csharp/csharp_config.feature (hunk 1 above)
Comment (rich-bugsnag) [reply features/csharp/csharp_config.feature:119]: i'll consider merging them in the next PR 👍

File: features/csharp/csharp_config.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (twometresteve) [reply features/csharp/csharp_config.feature:41]: That's the kind of positive attitude that I like.

**Pull request merged**